### ISSUE TITLE
Review "tidy evaluation" in docs

### ISSUE THREAD

**Issue by hadley** [tidyverse/dplyr#3953]
Currently, filter has:

> These arguments are automatically quoted and evaluated in the context of the data frame. 
> They support unquoting and splicing. See vignette("programming") for an introduction to 
> these concepts.

This is suboptimal because it doesn't explicitly say "tidy evaluation" anywhere, and I think we should be moving to deprecate the programming vignette in favour of https://tidyeval.tidyverse.org. For now, I think we should leave the vignette as is, but link to the in-progress book.

**Comment by romainfrancois** [comment]
There are a few mentions of the vignette. Should we replace all with links to the tidy eval book ? or is this case by case ?

```r
> fs::dir_ls("R", regexp = "[rR]$") %>% 
+   map(readr::read_lines) %>% 
+   map(stringr::str_subset, "vignette..programming") %>% 
+   purrr::keep(~length(.) > 0)
$`R/colwise.R`
[1] "#'   `vignette(\"programming\")` for an introduction to these concepts." "#'   `vignette(\"programming\")` for an introduction to these concepts."

$`R/count-tally.R`
[1] "#'   `vignette(\"programming\")` for an introduction to these concepts."

$`R/funs.R`
[1] "#'  `vignette(\"programming\")` for an introduction to these concepts."

$`R/manip.r`
[1] "#'   splicing. See `vignette(\"programming\")` for an introduction to"   "#'   splicing. See `vignette(\"programming\")` for an introduction to"  
[3] "#'   splicing. See `vignette(\"programming\")` for an introduction to"   "#'   splicing. See `vignette(\"programming\")` for an introduction to"  
[5] "#'   `vignette(\"programming\")` for an introduction to these concepts." "#' argument. See `vignette(\"programming\")` for more information."     

$`R/sample.R`
[1] "#'   `vignette(\"programming\")` for an introduction to these concepts."

$`R/top-n.R`
[1] "#'   `vignette(\"programming\")` for an introduction to these concepts."
```

**Comment by hadley** [comment]
Yeah, I think they should all point to the book.

**Comment by hadley** [comment]
Fixed by #4684
